### PR TITLE
MSC2663: Errors for dealing with non-existent push rules

### PR DESCRIPTION
**Pull request by reivilibre**
[Rendered](https://github.com/matrix-org/matrix-doc/blob/rei/pushrules_define_enabled/proposals/2663-errors-nonexistent-push-rules.md)

Signed-off-by: Olivier Wilkinson (reivilibre) <olivier@librepush.net>

### CODE REVIEW

**Comment by richvdh** [comment]
inexistent isn't a word. You might say... it's inexistent.

**Comment by reivilibre** [comment]
It's not clear to me that 'inexistent' isn't a word; it features on https://www.wordnik.com/words/inexistent .

However, it's not in the Cambridge dictionary so I suppose I will switch over :).

**Comment by richvdh** [comment]
synapse implementation is https://github.com/matrix-org/synapse/pull/7796

**Comment by richvdh** [comment]
I'm going to propose FCP on this even though the impl hasn't landed yet, because it's pretty trivial

**Comment by richvdh** [comment]
@mscbot fcp merge

File: proposals/2663-errors-nonexistent-push-rules.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+**Status code 404:**
+
+The push rule does not exist.
+
+**Example**
+```json
+{
+  "errcode": "M_NOT_FOUND"
+}
```
Comment (turt2live) [inline proposals/2663-errors-nonexistent-push-rules.md:40]: Hopefully with a message too when it comes to speccing.

**Comment by turt2live** [comment]
Spec PR: https://github.com/matrix-org/matrix-doc/pull/2796

**Comment by turt2live** [comment]
Merged 🎉

**Pull request merged**